test/python/circuit/test_circuit_registers.py

```diff
@@ -74,3 +74,81 @@ def test_cregs_circuit(self):
         self.assertEqual(len(c_regs), 3)
         self.assertEqual(c_regs[0], cr1)
         self.assertEqual(c_regs[1], cr2)
+
+    def test_basic_slice(self):
+        """simple slice test"""
+        qr = QuantumRegister(5)
+        cr = ClassicalRegister(5)
+        self.assertEqual(len(qr[0:3]), 3)
+        self.assertEqual(len(cr[0:3]), 3)
+
+    def test_apply_gate_to_slice(self):
+        """test applying gate to register slice"""
+        qr = QuantumRegister(10)
+        cr = ClassicalRegister(10)
+        qc = QuantumCircuit(qr, cr)
+        qc.x(qr[0])
+        qc.y(qr)
+        qc.x(qr[0:2])
+        qc.h(qr[slice(0, 9, 2)])
+        self.log.info(qc.qasm())
+
+    def test_apply_barrier_to_slice(self):
+        """test applying barrier to register slice"""
+        qr = QuantumRegister(10)
+        cr = ClassicalRegister(10)
+        qc = QuantumCircuit(qr, cr)
+        qc.barrier(qr)
+        qc.barrier(qr[0:2])
+        self.log.info(qc.qasm())
+
+    def test_apply_ccx_to_slice(self):
+        """test applying ccx to register slice"""
+        qcontrol = QuantumRegister(10)
+        qcontrol2 = QuantumRegister(10)
+        qtarget = QuantumRegister(5)
+        qtarget2 = QuantumRegister(10)
+        qc = QuantumCircuit(qcontrol, qtarget, qcontrol2, qtarget2)
+        qc.ccx(qcontrol[1::2], qcontrol[0::2], qtarget)
+        qc.barrier(qcontrol)
+        qc.ccx(qcontrol[2:0:-1], qcontrol[4:6], qtarget[0:2])
+        qc.barrier(qcontrol)
+        qc.ccx(qcontrol, qcontrol2, qtarget2)
+        qc.barrier(qcontrol)
+        qc.ccx(qcontrol[0:2], qcontrol[2:4], qcontrol[5:7])
+        self.log.info(qc.qasm())
```

Comment:
can you assert something in this test too?

Reply:
yes